horovod/common/operations.cc

```diff
@@ -1242,220 +1288,26 @@ void BackgroundThreadLoop(HorovodGlobalState& state) {
     state.cycle_time = std::atof(horovod_cycle_time);
   }
 
+  // Set flag for hierarchical allreduce. Ignore if Horovod is running on a
+  // single node.
+  auto horovod_hierarchical_allreduce =
+      std::getenv("HOROVOD_HIERARCHICAL_ALLREDUCE");
+  if (horovod_hierarchical_allreduce != nullptr &&
+      std::atoi(horovod_hierarchical_allreduce) > 0 && cross_size > 1) {
```

Comment:
Wondering if [strtol](https://en.cppreference.com/w/c/string/byte/strtol) might be safer here.